other/update_unicode_data.php

```diff
@@ -211,37 +477,336 @@
+			break;
+	}
+
+	if (!isset($joining_type))
+		continue;
+
+	$char_scripts = $char_data['&#x' . substr($fields[0], 0, strcspn($fields[0], '.')) . ';']['scripts'];
+
+	if (empty($char_scripts))
+		continue;
+
+	foreach ($char_scripts as $char_script) {
+		if (!isset($funcs['utf8_regex_joining_type']['data'][$char_script]['stats']))
+			$funcs['utf8_regex_joining_type']['data'][$char_script]['stats'] = $script_stats[$char_script];
+
+		if (!isset($funcs['utf8_regex_joining_type']['data'][$char_script][$joining_type]))
+			$funcs['utf8_regex_joining_type']['data'][$char_script][$joining_type] = array();
+
+		$funcs['utf8_regex_joining_type']['data'][$char_script][$joining_type][] = '\\x{' . str_replace('..', '}-\\x{', $fields[0]) . '}';
+	}
+}
+// This sort works decently well to ensure widely used scripts are ranked before rare scripts.
+uasort($funcs['utf8_regex_joining_type']['data'], function($a, $b) {
+	if ($a['stats']['age'] == $b['stats']['age']) {
+		return $b['stats']['count'] - $a['stats']['count'];
+	} else {
+		return $a['stats']['age'] - $b['stats']['age'];
+	}
+});
+foreach ($funcs['utf8_regex_joining_type']['data'] as $char_script => $joining_types) {
+	unset($funcs['utf8_regex_joining_type']['data'][$char_script]['stats'], $joining_types['stats']);
+
+	// If the only joining type in this script is transparent, we don't care about it.
+	if (array_keys($joining_types) === array('Transparent')) {
+		unset($funcs['utf8_regex_joining_type']['data'][$char_script]);
+		continue;
+	}
+
+	foreach ($joining_types as $joining_type => $value) {
+		sort($value);
+		$funcs['utf8_regex_joining_type']['data'][$char_script][$joining_type] = implode('', $value);
+	}
+}
+
+// 2. Indic scripts like Devanagari.
+foreach (file($unicode_data_url . '/IndicSyllabicCategory.txt') as $line) {
+	$line = substr($line, 0, strcspn($line, '#'));
+
+	if (strpos($line, ';') === false)
+		continue;
+
+	$fields = explode(';', $line);
+
+	foreach ($fields as $key => $value) {
+		$fields[$key] = trim($value);
+	}
+
+	$insc = $fields[1];
+
+	if (!in_array($insc, array('Virama', 'Vowel_Dependent')))
+		continue;
+
+	$char_scripts = $char_data['&#x' . substr($fields[0], 0, strcspn($fields[0], '.')) . ';']['scripts'];
+
+	if (empty($char_scripts))
+		continue;
+
+	foreach ($char_scripts as $char_script) {
+		if (!isset($funcs['utf8_regex_indic']['data'][$char_script]['stats']))
+			$funcs['utf8_regex_indic']['data'][$char_script]['stats'] = $script_stats[$char_script];
+
+		if (!isset($funcs['utf8_regex_indic']['data'][$char_script][$insc]))
+			$funcs['utf8_regex_indic']['data'][$char_script][$insc] = array();
+
+		$funcs['utf8_regex_indic']['data'][$char_script][$insc][] = '\\x{' . str_replace('..', '}-\\x{', $fields[0]) . '}';
+	}
+}
+// Again, sort commonly used scripts before rare scripts.
+uasort($funcs['utf8_regex_indic']['data'], function($a, $b) {
+	if ($a['stats']['age'] == $b['stats']['age']) {
+		return $b['stats']['count'] - $a['stats']['count'];
+	} else {
```

Comment:
ahem...er... this isn't Allman

Reply:
I don't understand this comment, @live627.

Reply:
Oh, you just mean the indentation style. It is true, I just used my text editor's automatic PSR-2 styling when I first made this file for private use, and then I never bothered to change it. I suppose I might as well...

Reply:
There, done.